Box.V2/Request/ReusableContent.cs

```diff
@@ -0,0 +1,35 @@
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Box.V2.Request
+{
+    class ReusableContent : HttpContent
+    {
+        private HttpContent _innerContent;
+
+        public ReusableContent(Stream stream)
+        {
+            _innerContent = new StreamContent(stream);
+        }
+
+        protected override async Task SerializeToStreamAsync(Stream stream, TransportContext context)
+        {
+            await _innerContent.CopyToAsync(stream);
```

Comment:
I do feel like it's not ideal to do this copy operation.  I don't fully understand it, TBH.  When exactly is this method called? When does the `stream` property get disposed?  Can we dispose `stream` at the end of this method to free up memory sooner and explicitly?

Maybe the token method you mentioned would be "safer."  Where would that check need to be inserted?

Good job researching this and coming up with an otherwise elegant solution!

Reply:
`SerializeToStreamAsync` is called internally by the `HttpClient` during `sendAsync(..)` method. We cannot dispose `_innerContent` yet as it will be used during the retry in case the request fails (`stream` should be disposed by `httpClient`). If we don't copy this, the `_innerContent` will be disposed internally and retry will fail. This is the behaviour of the `HttpClient` (see links in description) that dispose the resources it's not the owner of.

The token method I mentioned can solve stream issue in some cases and save memory by preventing copying, but still they are some cases that the retry would fail. By calling token just before the upload request, we can ensure that it will be valid. 401 code should be no longer present. There is still a possibility that the request will fail for different reason (e.g. 500 code from server). In this case the Stream will be disposed and retry will fail every time.  

The solution is not perfect, but I don't see any alternative if we want to stick to the current `HttpClient` and solve the retry problem.

Reply:
Makes sense!  Thanks!